src/client/form/pw-input-checkbox.ts

```diff
@@ -81,39 +80,74 @@ export function pwInputCheckbox<P extends keyof typeof routes>(
     autocomplete=${ifDefined(autocomplete)}
     .task=${task}
     .initial=${initial}
-    .defaultValue=${defaultValue}
-    .trueValue=${trueValue}
-    .falseValue=${falseValue}
     .resettable=${resettable}
   ></pw-input-checkbox>`;
 }
 
-export class PwInputCheckbox<P extends keyof typeof routes> extends PwInput<
+export class PwInputCheckbox<
+  P extends keyof typeof routes,
+  RESETTABLE extends boolean
+> extends PwInput<
   P,
   boolean | undefined,
+  RESETTABLE,
+  boolean,
   HTMLInputElement
 > {
   static override get properties() {
     return {
-      trueValue: { attribute: false },
-      falseValue: { attribute: false },
       ...super.properties,
     };
   }
 
-  falseValue!: boolean | undefined;
-  trueValue!: boolean | undefined;
+  get inner() {
+    return html`
+    <div>
+      <input
+        ${ref(this.input)}
+        type=${this.type}
+        name=${this.name}
+        .checked=${live(this.get(this.pwForm.formData))}
+        class="form-check-input ${this.task.render({
+          pending: () => noChange,
+          complete: (v) =>
+            !v.success &&
+            v.error.issues.find(
+              (i) => JSON.stringify(i.path) == JSON.stringify(this.name)
+            ) !== undefined
+              ? "is-invalid"
+              : "is-valid",
+          initial: () => "",
+        })}"
+        id=${this.randomId}
+        aria-describedby="${this.randomId}-feedback"
+        autocomplete=${ifDefined(this.autocomplete)}
+        ?disabled=${
+          !this.enabled &&
+          (this.disabled ||
+            this.task.render({
+              complete: () => false,
+              pending: () => true,
+              initial: () => false,
+            }))
+        }
+      ></input>
+  ${
+    this.label !== null
+      ? html`<label for=${this.randomId} class="form-check-label"
+          >${this.label}</label
+        >`
+      : undefined
+  }  ${this.inner2}     </div>
+  `;
+  }
 
   mypwinputchangeDispatcher = () => {
     if (!this.input.value) {
       throw new Error();
     }
 
-    this.inputValue = this.input.value.checked;
```

Comment:
This duplication of the same data (inputValue and the value in formData) was the main reason for refactoring.

To refactor this I believe we need the current approach as otherwise we can't to this correctly in all cases.